0.12: why upgrade to npm 3?

this breaks a bunch of things in my builds that use to work before, and there's no way to revert because you haven't versioned this change. 

I'm seeing  ``npm ERR! code EXDEV`` and ``npm WARN EBUNDLEOVERRIDE Replacing...`` messages that break my installation. when googling the first error, it seems the fix is to upgrade to node 4? why is this in a 0.12 Dockerfile then?

please either revert this change, or version it so we can choose npm v2 or npm v3.  those of us on node 0.12 don't want breaking changes, please leave this to node 5.

--- a/0.12/Dockerfile
+++ b/0.12/Dockerfile
@@ -14,15 +14,12 @@ RUN set -ex \
   done
 
 ENV NODE_VERSION 0.12.8
-ENV NPM_VERSION 3.4.1
 
 RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
 	&& curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
 	&& gpg --verify SHASUMS256.txt.asc \
 	&& grep " node-v$NODE_VERSION-linux-x64.tar.gz\$" SHASUMS256.txt.asc | sha256sum -c - \
 	&& tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
-	&& rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc \
-	&& npm install -g npm@"$NPM_VERSION" \
-	&& npm cache clear
+	&& rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc
 
 CMD [ "node" ]

--- a/0.12/slim/Dockerfile
+++ b/0.12/slim/Dockerfile
@@ -14,7 +14,6 @@ RUN set -ex \
   done
 
 ENV NODE_VERSION 0.12.8
-ENV NPM_VERSION 3.4.1
 
 RUN buildDeps='curl ca-certificates' \
 	&& set -x \
@@ -26,8 +25,6 @@ RUN buildDeps='curl ca-certificates' \
 	&& grep " node-v$NODE_VERSION-linux-x64.tar.gz\$" SHASUMS256.txt.asc | sha256sum -c - \
 	&& tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
 	&& rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc \
-	&& apt-get purge -y --auto-remove $buildDeps \
-	&& npm install -g npm@"$NPM_VERSION" \
-	&& npm cache clear
+	&& apt-get purge -y --auto-remove $buildDeps
 
 CMD [ "node" ]

--- a/0.12/wheezy/Dockerfile
+++ b/0.12/wheezy/Dockerfile
@@ -14,15 +14,12 @@ RUN set -ex \
   done
 
 ENV NODE_VERSION 0.12.8
-ENV NPM_VERSION 3.4.1
 
 RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
 	&& curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
 	&& gpg --verify SHASUMS256.txt.asc \
 	&& grep " node-v$NODE_VERSION-linux-x64.tar.gz\$" SHASUMS256.txt.asc | sha256sum -c - \
 	&& tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
-	&& rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc \
-	&& npm install -g npm@"$NPM_VERSION" \
-	&& npm cache clear
+	&& rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc
 
 CMD [ "node" ]

